### PR TITLE
V3

### PR DESCRIPTION
**Pull request by jr0d**
merge v3 into master

### CODE REVIEW

File: internal/config.go [hunk 1]
```diff
@@ -206,7 +210,7 @@ func (c *Config) Validate() {
 		log.Infoln("for better security, \"secret\" should ideally be 32 bytes or longer")
 	}
 
-	if c.ProviderUri == "" || c.ClientId == "" || c.ClientSecret == "" {
+	if c.ProviderURI == "" || c.ClientID == "" || c.ClientSecret == "" {
```
Comment (fdevillard) [inline internal/config.go:213]: As the `EncryptionKeyString` is now required, I'd suggest to add a validation step for it here.

**Review by fdevillard — COMMENTED** [review]
👋

First of all, thank you for your contributions :) 

I'm migrating the middleware from the original repo to this fork and new version. It looks like this additional validation step can be useful 🙃

File: README.md [hunk 2]
```diff
@@ -19,3 +23,10 @@ The instructions for [`thomseddon/traefik-forward-auth`](https://github.com/thom
 - Returns 401 rather than redirect to OIDC Login if an unauthenticated request is not for HTML (e.g. AJAX calls, images).
 - Sends a username cookie as well
 - If `auth-host` is set and `cookie-domains` is not set, traefik-forward-auth will redirect any requests using other hostnames to `auth-host`. Set `auth-host` to the OIDC redirect host to ensure that use of the IP or other DNS names will be redirected and get a suitable cookie.
+
+## Upgrading from 2.x version to 3.0 (Breaking Changes):
+
+- config `session-key` (`SESSION_KEY` env) is now called `encryption-key` (`ENCRYPTION_KEY` env) and is `REQUIRED`
+- config `groups-session-name` (`GROUPS_SESSION_NAME`) is deprecated as both email and groups are part of the single cookie `cookie-name` (`COOKIE_NAME` env)
```
Comment (fdevillard) [inline README.md:30]: Hum, the `cookie-name` is probably a typo and should be replaced with `user-cookie-name`.

**Comment by dkoshkin** [comment]
Closing in-favor of https://github.com/mesosphere/traefik-forward-auth/pull/54

**Pull request closed without merging**